tests/integration/src/test/scala/mesosphere/marathon/integration/facades/MesosFacade.scala

```diff
@@ -121,4 +134,20 @@ class MesosFacade(val url: String, val waitTime: FiniteDuration = 30.seconds)(im
   def teardown(frameworkId: String): HttpResponse = {
     result(request(Post(s"$url/teardown", HttpEntity(s"frameworkId=$frameworkId"))), waitTime).value
   }
+
+  /**
+    * Mark agent as gone using v1 operator API
+    *
+    * @param agentId
+    * @return Right(Done) on success, Left(errorString) otherwise
+    */
+  def markAgentGone(agentId: String): RestResult[Done] = {
+    val response = result(request(Post(s"$url/api/v1", Json.obj(
+      "type" -> "MARK_AGENT_GONE",
+      "mark_agent_gone" -> Json.obj(
+        "agent_id" -> Json.obj("value" -> agentId))))), waitTime)
+    response.map { _ =>
+      Done
```

Comment:
What is the motivation for returning `Done` and not `HttpResponse`? The later would help debugging.

Reply:
The entity is actually empty if it is successful. Using `RestResult[SuccessType]` is a consistent pattern with the rest of these methods.

Reply:
Hm, this is funny. `treadown` is actually not using the common pattern of `RestResult`. That's why I was confused 🙂